demo-ling/stock_param_functions.r

```diff
@@ -0,0 +1,47 @@
+## ADDITIONAL FUNCTIONS TO HELP WITH THE SETUP SCRIPTS
+
+## Function to insert stock/species name into g3 parameter
+g3_stock_param <- function(sname, param, lower=NULL, upper=NULL, xby=NULL, sep0="."){
```

Comment:
This should take in the stock object, not the stock name. Then you can pull the name out of that using ``stock$name``.

As I've mentioned somewhere else, allowing ``sep0`` to be configurable just makes your life harder long-term. More consistent naming saves a lot of brain time.

Generally the fact that you need this is annoying (but it does make sense currently).  ``g3_param()`` did have the ability to make names out of several parts like this, but I accidentally removed it.

Reply:
I modified the g3_stock_param function in commit 87980ee. It is now a smaller function that is called by bounded_param that adds the bounds to the parameter reference:

`  model_params <- list('K' = list(lower = ..., upper = ...))
  ## For ling.K
  bounded_param(stock_species(ling_imm), 'K', model_params)
  ## For ling_imm.K
  bounded_param(ling_imm$name, 'K', model_params)`  

Both the xby and sep0 parameters have been removed :)

However, I've kept a character string as the input (now `stock_name = ...`) as it is easily to switch between a stock-specific parameter `stock_object$name` ('ling_imm') and a parameter that is consistent across stock `species_stock(stock_object)` ('ling')